src/libexpr/primops/context.cc

```diff
@@ -67,10 +61,51 @@ static void prim_unsafeDiscardOutputDependency(EvalState & state, const PosIdx p
 static RegisterPrimOp primop_unsafeDiscardOutputDependency({
     .name = "__unsafeDiscardOutputDependency",
     .arity = 1,
+    .doc = R"(
+      TODO! Old comment is not good.
+
+      Sometimes we want to pass a derivation path (i.e. `pkg.drvPath`) to a builder without causing the derivation to be built
+      (for instance, in the derivation that builds NARs in nix-push, when doing source-only deployment).
```

Comment:
```suggestion
      (for instance, when building a deployment script that builds already-instantiated derivations on the deployment target; a source based deployment that's independent of fetcher configuration among other things).
```

An undocumented deployment method, as far as I know. This should really be a reference to some definition, but we don't think we have that yet.

(and it's not necessarily source-only if the deployment target has substituters)